subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskCreationBuildOperationIntegrationTest.groovy

```diff
@@ -285,15 +303,23 @@ class TaskCreationBuildOperationIntegrationTest extends AbstractIntegrationSpec
     }
 
     private <T extends BuildOperationType<?, ?>> BuildOperationRecord verifyTaskDetails(Map<String, ?> expectedDetails, Class<T> type, Spec<? super BuildOperationRecord> spec) {
-        def op = buildOperations.only(type, spec)
-        verifyTaskDetails(expectedDetails, op)
-        op
+        def ops = buildOperations.all(type, spec)
+        assert !ops.empty
+        if (type == RealizeTaskBuildOperationType && GradleContextualExecuter.configCache) {
+            // When using load after store, the task will be realized twice:
+            //  - at configuration time
+            //  - after loading from the configuration cache
+            assert ops.size() <= 2
```

Comment:
❔ If it is always realized twice and we already assert `executer.configCache`, why not check for `== 2` ?

Reply:
I added an explanation.